### PR TITLE
Add fixes to vLLM install and triton kernel optimization

### PR DESCRIPTION
**Pull request by peterjunpark**
Remove confusing information from **Fine-tuning LLMs and inference optimization**

### CODE REVIEW

File: docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst
Comment (hongxiayang) [inline docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:33]: we can add the following two lines before the `docker build` command:
```
git clone https://github.com/vllm-project/vllm.git
cd vllm
```

**Review by hongxiayang — APPROVED** [review]
Approve with comment. For more clarity, please add those two lines. Thank you @peterjunpark

File: docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst
Comment (peterjunpark) [reply docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:33]: Got this already
```      
      git clone https://github.com/vllm-project/vllm.git
      cd vllm
      docker build -f Dockerfile.rocm -t vllm-rocm .
```

File: docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst
Comment (seungrokj) [inline docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:46]: vllm-rocm -> vllm

File: docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst
Comment (seungrokj) [inline docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:90]: vllm-rocm -> vllm

File: docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst
Comment (hongxiayang) [reply docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:46]: > vllm-rocm -> vllm

vllm-rocm is the docker image name built from previous step.

File: docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst
Comment (hongxiayang) [reply docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:33]: thanks. you are correct.

File: docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst
Comment (seungrokj) [inline docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:129]: below line 157: 164
1. To install the TGI Docker image, run the following commands.
   .. code-block:: shell
      # Install from Dockerfile
      git clone https://github.com/huggingface/text-generation-inference.git -b mi300-compat    
      cd text-generation-inference
      docker build . -f Dockerfile.rocm

->
1. To launch the TGI Docker instance, run the following commands.
   .. code-block:: shell
      # Launch the TGI Docker in the host machine
docker run --name tgi --rm -it --cap-add=SYS_PTRACE --security-opt seccomp=unconfined \
    --device=/dev/kfd --device=/dev/dri --group-add video --ipc=host --shm-size 256g \
    --net host -v $PWD:/data \
    --entrypoint "/bin/bash" \
    --env HUGGINGFACE_HUB_CACHE=/data \
    ghcr.io/huggingface/text-generation-inference:latest-rocm

Comment (seungrokj) [inline docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:129]: Also, below line 171

2. Launch a model using TGI server on a single accelerator.
->
2. Inside the container, launch a model using TGI server on a single accelerator.

Comment (peterjunpark) [reply docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:129]: done

Comment (peterjunpark) [reply docs/how-to/llm-fine-tuning-optimization/llm-inference-frameworks.rst:129]: done

**Comment by peterjunpark** [comment]
Need approval from @ROCm/rocm-documentation as code owner

**Review by seungrokj — APPROVED** [review]
LGTM

**Review by hongxiayang — APPROVED** [review]
lgtm. thanks

**Review by amitkumar-amd — APPROVED** [review]

**Pull request merged**
